problem_builder/tests/unit/test_instructor_tool.py

```diff
@@ -0,0 +1,226 @@
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree: Should only include root block
+        self.assertEqual(len(block_tree), 1)
+
+    def test_student_view_block_tree_eligible(self):
+        """
+        Check if `student_view` method correctly marks MCQ, Rating, and
+        Answer blocks as eligible when building tree for render
+        context.
+
+        A block is eligible if its type is one of {'pb-mcq', 'pb-rating', 'pb-answer'}.
+        """
+        # Pretend that all blocks in self.block.children are eligible:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = random.choice(['pb-mcq', 'pb-rating', 'pb-answer'])
+
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree: All blocks should be included
+        num_blocks = len(self.block.children) + 1
+        self.assertEqual(len(block_tree), num_blocks)
+
+        # Check if all blocks are eligible:
+        self.assertTrue(all(block['eligible'] for block in block_tree))
+
+    def test_student_view_block_tree_ineligible(self):
+        """
+        Check if `student_view` method correctly marks blocks that aren't
+        MCQ, Rating, or Answer blocks as ineligible when building tree
+        for render context.
+        """
+        # Pretend that none of the blocks in self.block.children are eligible:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = 'pb-dummy-type'
+
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree: All blocks should be included (they are not of type 'pb-choice')
+        num_blocks = len(self.block.children) + 1
+        self.assertEqual(len(block_tree), num_blocks)
+
+        # Check if all blocks are ineligible:
+        self.assertTrue(all(not block['eligible'] for block in block_tree))
+
+    def test_student_view_old_style_keys_not_supported(self):
+        """
+        Check if student_view correctly handles scenarios in which
+        old-style keys are not supported.
+        """
+        self.block.scope_ids.def_id = Mock()
+        self.block.scope_ids.def_id.block_type.side_effect = AttributeError()
+
+        try:
+            self.block.student_view(context={})
+        except AttributeError:
+            self.fail('student_view breaks if old-style keys are not supported.')
```

Comment:
@itsjeyd I'm totally confused by this test. So it should fail if `def_id.block_type` is accessed, right? How is that connected to old-style keys? Some more verbose docstring would be nice to have